File: WeddingWebsite-Backend/WeddingInvites/Domain/Invite.cs
using System.ComponentModel.DataAnnotations;

namespace WeddingInvites.Domain;

public class Invite : BaseEntity
{
    public string Name { get; set; }
    
    public IEnumerable<Guest> Guests { get; set; } = new List<Guest>();
    
    [MaxLength(10)]
    public string PublicCode { get; set; }
    
    public bool RsvpCompleted { get; set; }
}

File: WeddingWebsite-Angular/src/app/models/invite.ts
import { BaseEntity } from './base-entity';
import { Guest } from './guest';
import { JsonPreserveWrapper } from './json-preserve-wrapper';

export interface Invite extends BaseEntity {
  name: string;
  guests: JsonPreserveWrapper<Guest>;
  publicCode: string;
  rsvpCompleted: boolean;
}
